code/modules/awaymissions/mission_code/ruins/deepstorage.dm

```diff
@@ -0,0 +1,380 @@
+	if(charging && yes)
+		if(isliving(A))
+			var/mob/living/L = A
+			L.visible_message("<span class='danger'>[src] slams into [L]!</span>", "<span class='userdanger'>[src] tramples you into the ground!</span>")
+			forceMove(get_turf(L))
+			L.apply_damage(istype(src, /mob/living/simple_animal/hostile/megafauna/bubblegum/hallucination) ? 15 : 30, BRUTE)
+			playsound(get_turf(L), 'sound/effects/meteorimpact.ogg', 100, TRUE)
+			shake_camera(L, 4, 3)
+			shake_camera(src, 2, 3)
+	..()
+
+/mob/living/simple_animal/hostile/megafauna/fleshling/Destroy()
+	handle_dying()
+	return ..()
+
+/mob/living/simple_animal/hostile/megafauna/fleshling/proc/handle_dying()
+	if(!boss_killed)
+		boss_killed = TRUE
+
+/mob/living/simple_animal/hostile/megafauna/fleshling/death(gibbed)
+	if(can_die() && !boss_killed)
+		unlock_blast_doors(DS_BOSS_STORAGE)
+	return ..(gibbed)
+
+/mob/living/simple_animal/hostile/megafauna/fleshling/proc/unlock_blast_doors(target_id_tag)
+	for(var/obj/machinery/door/poddoor/P in GLOB.airlocks)
+		if(P.density && P.id_tag == target_id_tag && P.z == z && !P.operating)
+			P.open()
+
+/mob/living/simple_animal/hostile/spaceinfected
+	name = "Infected"
+	desc = "A reanimated corpse, wandering around aimlessly."
+	icon = 'icons/mob/simple_human.dmi'
+	icon_state = "spaceinfected"
+	icon_living = "spaceinfected"
+	mob_biotypes = MOB_ORGANIC | MOB_HUMANOID
+	speak_chance = 1
+	turns_per_move = 3
+	death_sound = 'sound/effects/bodyfall1.ogg'
+	speed = 0
+	maxHealth = 150
+	health = 150
+	melee_damage_lower = 20
+	melee_damage_upper = 20
+	attacktext = "hits"
+	attack_sound = 'sound/effects/blobattack.ogg'
+	del_on_death = TRUE
+	sentience_type = SENTIENCE_OTHER
+	footstep_type = FOOTSTEP_MOB_SHOE
+	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	minbodytemp = 0
+	loot = list(/obj/effect/decal/cleanable/blood/innards,
+			/obj/effect/decal/cleanable/blood,
+			/obj/effect/gibspawner/generic,
+			/obj/effect/gibspawner/generic)
+
+/mob/living/simple_animal/hostile/spaceinfected/ListTargetsLazy()
+	return ListTargets()
+
+/mob/living/simple_animal/hostile/spaceinfected/Aggro()
+	. = ..()
+	if(target)
+		playsound(loc, 'sound/voice/zombie_scream.ogg', 70, TRUE)
+
+/mob/living/simple_animal/hostile/spaceinfected/Move(atom/newloc)
+	if(ischasm(newloc)) // as this place filled with chasms, they shouldn't randomly fall in while wandering around
+		return FALSE
+	return ..()
+
+/mob/living/simple_animal/hostile/spaceinfected/default
+
+/mob/living/simple_animal/hostile/spaceinfected/default/Initialize(mapload)
+	. = ..()
+	var/loot_num = rand(1,100)
+	switch(loot_num)
+		if(1 to 10)
+			loot = list(/obj/item/salvage/ruin/nanotrasen,
+				/obj/effect/decal/cleanable/blood/innards,
+				/obj/effect/decal/cleanable/blood,
+				/obj/effect/gibspawner/generic,
+				/obj/effect/gibspawner/generic)
+
+		if(11 to 30)
```

Comment:
The rand range is 1-100 but you you only check for values from 1-30, is that intentional?

Reply:
70% of the time mobs shouldn't drop any loot, hence their loot variable doesn't get overridden